viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -98,6 +114,7 @@ config from_viam_carto_config(viam_carto_config vcc) {
     if (vcc.map_rate_sec < 0) {
         throw VIAM_CARTO_MAP_RATE_SEC_INVALID;
     }
+    c.component_reference = c.sensors[0];
```

Comment:
We no longer take a component_reference from the vcc and instead infer it from the first element of the sensors list (as slam_service does today).